rsconnect/bundle.py

```diff
@@ -1062,6 +1062,15 @@ def fake_module_file_from_directory(directory: str):
     return join(directory, app_name + ".py")
 
 
+def is_python_enabled_on_server(connect_details):
+    """
+    Returns whether or not the Connect server has Python itself enabled.
+
+    :error: The Posit Connect server does not have Python enabled.
+    """
+    return any(connect_details["python"]["versions"])
```

Comment:
Here's another way to avoid the KeyError when the "python" key doesn't exist.

```suggestion
    return "python" in connect_details and "versions" in connect_details["python"]
```

Reply:
The reason we use `any` here is because "versions' is a list, so we also need to check whether it is empty.

Reply:
This is what connect_details  looks like:
connect_details = {'connect': '2023.02.0-dev+202-dirty', 'python': {'api_enabled': False, 'versions': []}, 'conda': {'supported': False}}

Reply:
That makes sense. 

The potential issue is that the key `python` may not exist in the object `connect_details`, which will result in a KeyError raise. This same logic is also true for the `versions` key.

The following snippet should avoid situations where the KeyError may be raised. 

```suggestion
    return any(connect_details.get("python", {}).get("versions", []))
```